crates/chia-bls/src/signature.rs

```diff
@@ -1203,6 +1216,27 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_scalar_multiply_large() {
+        let mut rng = StdRng::seed_from_u64(1337);
+        let mut data = [0; 4198];
+        rng.fill(data.as_mut_slice());
+        let seed: [u8; 32] = rng.gen();
+        let msg: [u8; 32] = rng.gen();
+        let sk = SecretKey::from_seed(&seed);
+        let mut g2 = sign(&sk, msg);
+        g2.scalar_multiply(&data);
+        assert_eq!(
+            hex::encode(g2.to_bytes()),
+            "
+            ae4d384a25c51283b8be8c6546d23e1555995c87fbcc0fe12169b63a052c4a1f
+            c9d3a020e8e010d4be619e3c0980a1f213b951fe75375012c5df6a690548a637
+            ef25f8da1e4f9c8f4d2062531ce688c040258a76543831abde774872e00af74b
+            "
+            .replace([' ', '\n'], "")
+        );
```

Comment:
you should have a comment with a reference where this test vector comes from, or you could turn it into a property-check, to ensure the result satisfies some expected properties of scalar multiplication